### ISSUE TITLE
Add warning for list_files function and related

### ISSUE THREAD

**Issue by ovidiul** [Automattic/VIP-Coding-Standards#704]
## What problem would the enhancement address for VIP?

I've recently been doing a code review and noticed the `list_files` function had no warning attached. On VIP Filesystem, the `list_files`, `scandir`, `opendir` will return empty result sets or false, so we should probably highlight this to customers to avoid unexpected results.

## Describe the solution you'd like
A warning should be added when the following functions are coded:
- list_files - WP
- scandir - PHP
- opendir - PHP

## What code should be reported as a violation?

```
$files = list_files( $folder, 2 );
```

**Comment by GaryJones** [comment]
These would need to go into https://github.com/Automattic/VIP-Coding-Standards/blob/develop/WordPressVIPMinimum/Sniffs/Functions/RestrictedFunctionsSniff.php

**Comment by GaryJones** [comment]
I've tested this on a VIP test site, and I'm getting the expected results for each of them, so I will close this out.

cc @yolih and @raamdev, who I think had worked on docs around this that may need to be reverted/updated.

**Comment by jrfnl** [comment]
@yolih @GaryJones I have a feeling there is some miscommunication going on here.

When I read @GaryJones' message, I get the impression he got proper correct file listings, not an empty array. @GaryJones is that correct ?
If so, that would mean the docs would need to be updated to remove the warning about those functions ?

**Comment by GaryJones** [comment]
Correct. On my VIP test site, the three functions worked as expected. I can get it verified, and if so, or docs need fixing.

**Comment by raamdev** [comment]
[The docs](https://docs.wpvip.com/technical-references/vip-go-files-system/media-uploads/) are referring to file listing functions on `wp-content/uploads/` paths on VIP, which utilize the VIP Filesystem. 

I've tested and confirmed that the following is true when those functions are used on `wp-content/uploads/`:

> [scandir()](https://www.php.net/manual/en/function.scandir.php), [list_files()](https://developer.wordpress.org/reference/functions/list_files/), and [opendir()](https://www.php.net/manual/en/function.opendir.php) will not work as expected, and will instead return either an empty array or false and trigger a PHP Warning.

I will reopen this issue with the caveat that we should produce a warning indicating that if those functions are being used on a path that points to the `wp-content/uploads/` directory, they will not work as intended.
